server/channels/app/import.go

```diff
@@ -45,20 +46,36 @@ func processAttachmentPaths(c request.CTX, files *[]imports.AttachmentImportData
 	if files == nil {
 		return nil
 	}
+
 	var ok bool
+	var multiErr *multierror.Error
 	for i, f := range *files {
 		if f.Path != nil {
-			path := filepath.Join(basePath, *f.Path)
+			originalPath := *f.Path
+
+			path, valid := imports.ValidateAttachmentPathForImport(originalPath, basePath)
+
 			*f.Path = path
+
+			if !valid {
+				multiErr = multierror.Append(multiErr, fmt.Errorf("invalid attachment path %q", originalPath))
+				continue
+			}
+
 			if len(filesMap) > 0 {
-				if (*files)[i].Data, ok = filesMap[path]; !ok {
-					return fmt.Errorf("attachment %q not found in map", path)
+				if (*files)[i].Data, ok = filesMap[*f.Path]; !ok {
```

Comment:
`*f.Path` has potentially changed here, but `filesMap` has not -- will we always trip this block if `ValidateAttachmentPathForImport` cleans some part of the path but doesn't outright reject it?

Reply:
That shouldn't be substantially different from before. `filesMap` expects the joined path, so we are passing our best attempt at that.

Reply:
Yep, that makes sense: in the happy path, the `filesMap` is already effectively populated using a key that joins the base path and `f.Path`, so once we join again, we should "find the same key". 

Maybe there's a way to avoid relying on `f.Name` in the first place? As in, if we trust the `map` completely, we can remove that fallback to the local file system and sidestep much of the impetus behind this PR?